webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -58,6 +58,21 @@ import com.normation.rudder.reports.ReportsDisabled
 import com.normation.rudder.domain.nodes.NodeState
 import com.normation.utils.Control.sequence
 
+
+object ReportingServiceUtils {
+
+  /*
+   * Build rule status reports from node reports, decide=ing which directive should be "skipped"
+   */
+  def buildRuleStatusReport(ruleId: RuleId, nodeReports: Map[NodeId, NodeStatusReport]): RuleStatusReport = {
+    val toKeep = nodeReports.values.flatMap( _.report.reports ).toList
```

Comment:
here, you should probaly add a filter on the ruleId, to ensure that nodestatusreports don't have extra rules that would polute it